### PR TITLE
HTML: fieldset legend should inherit alignment

### PR DESCRIPTION
**Pull request by zcorpan**
See https://github.com/whatwg/html/issues/3913.

### CODE REVIEW

**Review by annevk — APPROVED** [review]

File: html/rendering/non-replaced-elements/the-fieldset-element-0/legend-align.html [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+<div align=left>
+ <fieldset><legend>x</legend></fieldset>
+</div>
+<div align=center>
+ <fieldset><legend>x</legend></fieldset>
+</div>
+<div align=right>
+ <fieldset><legend>x</legend></fieldset>
+</div>
+<div style="text-align: center">
```
Comment (mstensho) [inline html/rendering/non-replaced-elements/the-fieldset-element-0/legend-align.html]: It doesn't seem right to me that text-align should affect the position of the legend block. It should only affect alignment of the text itself within lines.

Comment (zcorpan) [reply html/rendering/non-replaced-elements/the-fieldset-element-0/legend-align.html]: I guess we have a choice here. WebKit and Chromium position based on text-align, EdgeHTML and Gecko do not.

Comment (mstensho) [reply html/rendering/non-replaced-elements/the-fieldset-element-0/legend-align.html]: What WebKit / Chromium do should be considered a bug.

Comment (annevk) [reply html/rendering/non-replaced-elements/the-fieldset-element-0/legend-align.html]: Wait, doesn't Gecko also do so with a special text-align value?

Comment (zcorpan) [reply html/rendering/non-replaced-elements/the-fieldset-element-0/legend-align.html]: Gecko has a special text-align value for `<div align>` and `<table align>` etc, but, AIUI, not for `<legend align>`.

Setting `text-align: center` on a table does not center the table with auto margins. But setting `text-align: center` positions a legend in the center, which is what @mstensho thinks is a bug.

Comment (annevk) [reply html/rendering/non-replaced-elements/the-fieldset-element-0/legend-align.html]: @bzbarsky could you double check the above?

**Comment by zcorpan** [comment]
@mstensho PTAL

File: html/rendering/non-replaced-elements/the-fieldset-element-0/legend-align.html [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+for (const val of ['left', 'center', 'right']) {
+  test_align(`div[align=${val}] legend`, `legend[align=${val}]`);
+}
+
+test_align(`div[style="text-align: center"] legend`, `legend[align=left]`);
+test_align(`div[style="text-align: center"][align=center] legend`, `legend[align=center]`);
+test_align(`legend[style="margin: 0 auto"]`, `legend[align=center]`);
+test_align(`legend[style="margin: 0 0 0 auto"]`, `legend[align=right]`);
+test_align(`fieldset[dir=rtl] legend`, `legend[align=right]`);
+test_align(`fieldset[dir=rtl] legend[style="text-align: left"]`, `legend[align=left]`);
```
Comment (mstensho) [inline html/rendering/non-replaced-elements/the-fieldset-element-0/legend-align.html]: This legend should be right-aligned too.

Comment (zcorpan) [reply html/rendering/non-replaced-elements/the-fieldset-element-0/legend-align.html]: fixed

**Review by mstensho — APPROVED** [review]

**Comment by zcorpan** [comment]
@jugglinmike any idea why the stability checks failed? I tried rebasing on master, which still failed. Then restarted the jobs, still fail. :(

**Comment by jugglinmike** [comment]
Afraid not, @zcorpan. Since the error mentions the manifest version (`ManifestVersionMismatch`--stack trace below), I'm still inclined to think it's somehow related to gh-12563.  @foolip / @jgraham does this look at all familiar to you?

<details>
  <summary>Stack trace</summary>

    Traceback (most recent call last):
      File "./wpt", line 5, in <module>
        wpt.main()
      File "/home/travis/build/web-platform-tests/wpt/tools/wpt/wpt.py", line 132, in main
        rv = script(*args, **kwargs)
      File "/home/travis/build/web-platform-tests/wpt/tools/ci/check_stability.py", line 252, in run
        ignore_changes, skip_tests)
      File "/home/travis/build/web-platform-tests/wpt/tools/ci/check_stability.py", line 192, in get_changed_files
        manifest_path=manifest_path)
      File "/home/travis/build/web-platform-tests/wpt/tools/wpt/testfiles.py", line 212, in affected_testfiles
        wpt_manifest = load_manifest(manifest_path)
      File "/home/travis/build/web-platform-tests/wpt/tools/wpt/testfiles.py", line 193, in load
        wpt_manifest = manifest.load(wpt_root, manifest_path)
      File "/home/travis/build/web-platform-tests/wpt/tools/manifest/manifest.py", line 239, in load
        rv = Manifest.from_json(tests_root, json.load(f), types=types, meta_filters=meta_filters)
      File "/home/travis/build/web-platform-tests/wpt/tools/manifest/manifest.py", line 183, in from_json
        raise ManifestVersionMismatch
    tools.manifest.manifest.ManifestVersionMismatch

</details>

**Comment by zcorpan** [comment]
Restarting now made things green. Gonna guess that it was related to this:
```
gsnedders
did someone purge all the Travis caches?
gsnedders
it looks like someone did?
jgraham
Yes
```

**Pull request merged**